landlab/components/species_evolution/species_evolver.py

```diff
@@ -0,0 +1,673 @@
+    array([ True,  True,  True,  True,  True,  True,  True,  True,  True,
+            True,  True,  True,  True,  True,  True,  True,  True,  True,
+            True,  True,  True], dtype=bool)
+
+    Populate a taxon to the zone. The attribute, ``taxa_data_frame`` indicates
+    only the one taxon exists because we populated each zone with one taxon,
+    and only one zone exists.
+
+    >>> taxon = zc.populate_zones_uniformly(1)
+    >>> se.track_taxa(taxon)
+    >>> se.taxa_data_frame
+         appeared  latest_time  extant
+    uid
+    0           0            0    True
+
+    Force a change in the zone mask to demonstrate component functionality.
+    Here we begin a new time step where topography is uplifted by 200 that
+    forms a ridge trending north-south in the center of the grid.
+
+    >>> z[[3, 10, 17]] = 200
+    >>> z.reshape(mg.shape)
+    array([[   1.,    1.,    1.,  200.,    1.,    1.,    1.],
+           [   1.,    1.,    1.,  200.,    1.,    1.,    1.],
+           [   1.,    1.,    1.,  200.,    1.,    1.,    1.]])
+
+    The current elevation, the elevation following uplift, is represented here.
+
+    - - - ^ - - -       elevation:  - 1
+    - - - ^ - - -                   ^ 200
+    - - - ^ - - -
+
+    The updated zone mask is below.
+
+    . . . x . . .       key:    . node in zone mask
+    . . . x . . .               x node outside of zone mask
+    . . . x . . .
+
+    Run a step of both the ZoneController and SpeciesEvolver. Two zones exist
+    following this time step.
+
+    >>> delta_time = 1000
+    >>> zc.run_one_step(delta_time)
+    >>> se.run_one_step(delta_time)
+    >>> len(zc.zones) == 2
+    True
+
+    A additional zone was created because the zone mask was not continuous.
+
+    . . . ^ * * *       key:    . a zone
+    . . . ^ * * *               * another zone
+    . . . ^ * * *               ^ mountain range
+
+    The split of the initial zone triggered speciation. Taxon 0 became extinct
+    as it speciated to child taxa, 1 and 2.
+
+    >>> se.taxa_data_frame
+         appeared  latest_time  extant
+    uid
+    0           0         1000   False
+    1        1000         1000    True
+    2        1000         1000    True
+
+    The phylogeny model taxa is represented below. The number at the line tips
+    are the taxa identifiers.
+    ::
+
+                   ┌─ 1
+          0 ───────┤
+                   └─ 2
+            _________
+            0    1000
+              time
+    """
+    _name = 'SpeciesEvolver'
+
+    _info = {
+        "taxa__richness": {
+            "dtype": int,
+            "intent": "out",
+            "optional": False,
+            "units": "-",
+            "mapping": "node",
+            "doc": "The number of taxa at each node",
+        },
+    }
+
+    def __init__(self, grid, initial_time=0):
+        """Instantiate SpeciesEvolver.
+
+        Parameters
+        ----------
+        grid : ModelGrid
+            A Landlab ModelGrid.
```

Comment:
I think this is actually grid agnostic, as in one could use any grid type, right? If I'm wrong, then maybe state RasterModelGrid here.

Reply:
The component is grid agnostic. ZoneController is not. At some point I did indicate such. I must have overruled myself. Anyways, ZoneController grid parameter type is correct now.